packages/server/test/speckle-helpers/commitHelper.ts

```diff
@@ -73,84 +75,83 @@ export async function createTestObject(params: { projectId: string }) {
   })
 }
 
-const ensureObjectsFactory =
-  (deps: { db: Knex }) => async (commits: BasicTestCommit[]) => {
-    const { db } = deps
-    const createObject = createObjectFactory({
-      storeSingleObjectIfNotFoundFactory: storeSingleObjectIfNotFoundFactory({ db }),
-      storeClosuresIfNotFound: storeClosuresIfNotFoundFactory({ db })
-    })
+/**
+ * Ensure all commits have objectId set
+ */
+async function ensureObjects(commits: BasicTestCommit[]) {
+  const commitsWithoutObjects = commits.filter((c) => !c.objectId)
+  await Promise.all(
+    commitsWithoutObjects.map(async (c) => {
+      const projectDb = await getProjectDbClient({ projectId: c.streamId })
+      const createObject = createObjectFactory({
+        storeSingleObjectIfNotFoundFactory: storeSingleObjectIfNotFoundFactory({
+          db: projectDb
+        }),
+        storeClosuresIfNotFound: storeClosuresIfNotFoundFactory({ db: projectDb })
+      })
 
-    const commitsWithoutObjects = commits.filter((c) => !c.objectId)
-    await Promise.all(
-      commitsWithoutObjects.map((c) =>
-        createObject({
-          streamId: c.streamId,
-          object: { foo: 'bar' }
-        }).then((oid) => (c.objectId = oid))
-      )
-    )
-  }
+      return createObject({
+        streamId: c.streamId,
+        object: { foo: 'bar' }
+      }).then((oid) => (c.objectId = oid))
+    })
+  )
+}
 
 /**
  * Create test commits
  */
-export const createTestCommitsFactory =
-  (deps: { db: Knex }) =>
-  async (
-    commits: BasicTestCommit[],
-    options?: Partial<{ owner: BasicTestUser; stream: BasicTestStream }>
-  ) => {
-    const { db } = deps
-    const { owner, stream } = options || {}
+export async function createTestCommits(
```

Comment:
makes more sense for these to autoresolve the projectdb, instead of test writers having to do that manually beforehand